public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx

```diff
@@ -0,0 +1,257 @@
+                   // Change <= 20 count to display '<= 20'
+                   if (label <= 20) {
+                    return '&#8804; 20';
+                   }
+                   return label;
+                   },
+                   useHTML: true
+            };
+    }
+    baseOptions.series = series;
+    console.log(baseOptions);
+    this.setState({options: baseOptions});
+  }
+
+  prepSurveyCategoriesAndData() {
+    let data = [];
+    let cats = [];
+    let legendText = null;
+    // LOOP CREATES DYNAMIC CHART VARS
+    for (const a of this.props.genderAnalysis.results) {
+          // For normal Gender Analysis , the stratum2 is the gender . For ppi it is stratum5;
+          const analysisId = this.props.genderAnalysis.analysisId;
+          let analysisStratumName = null;
+          let toolTipHelpText = null;
+          let bsResult = null;
+          const color = '#2691D0';
+          let percentage = null;
+          let count;
+          let totalCount;
+          count = (a.countValue <= 20) ? '&le; 20' : a.countValue;
+          if (a.analysisStratumName === null) {
+            a.analysisStratumName = GENDER_STRATUM_MAP[a.stratum5];
+          }
+          analysisStratumName = a.analysisStratumName;
+          legendText = 'Sex Assigned At Birth, Selected Answered Count';
+          bsResult = this.props.genderCountAnalysis.results.
+                        filter(x => x.stratum2 === a.stratum5)[0];
+          totalCount = (bsResult.countValue <= 20) ? '&le; 20' : bsResult.countValue;
+                      percentage = Number(((a.countValue / bsResult.countValue) * 100).toFixed());
+                      toolTipHelpText = '<div class="chart-tooltip">' +
+                        '<strong> ' + count + '</strong> participants had ' + analysisStratumName +
+                        ' as sex assigned at birth with this survey answer and that is ' + '<strong>' +
+                        percentage + '% </strong>' + 'of the total count of ' +
+                        analysisStratumName + ' as sex assigned at birth that answered this ' +
+                        'survey question (total count ' +
+                        '= <strong> ' + totalCount + '</strong>) </div>';
+          data.push({
+            name: a.analysisStratumName
+            , y: a.countValue, color: color, sliced: true,
+            toolTipHelpText: toolTipHelpText, medicalConceptPercentage: percentage,
+            analysisId: analysisId
+          });
+          cats.push(a.analysisStratumName);
+        }
+        data = data.sort((a, b) => {
+          if (a.name > b.name) {
+            return 1;
+          }
+          if (a.name < b.name) {
+            return -1;
+          }
+          return 0;
+        }
+        );
+        cats = cats.sort((a, b) => {
+          if (a > b) {
+            return 1;
+          }
+          if (a < b) {
+            return -1;
+          }
+          return 0;
+        });
```

Comment:
Same as above, no need to reassign. Also if it's an array of strings, you can call with no function and it will sort alphabetically by default:
```suggestion
        cats.sort();
```

Reply:
Changed this.